### PR TITLE
Fix SFML Freetype links on Windows

### PR DESCRIPTION
**Pull request by SirLynix**

### CODE REVIEW

**Comment by SirLynix** [comment]
I don't understand that Linux error, but this isn't caused by this commit

**Comment by waruqi** [comment]
> I don't understand that Linux error, but this isn't caused by this commit

>checking for c++ includes(SFML/Graphics.hpp) 

The new test examples test more interface links and therefore trigger more hidden issues.

```lua
        if package:config("graphics") then
            assert(package:check_cxxsnippets({test = [[
                void test(int args, char** argv) {
                    sf::Text text;
                    text.setString("Hello world");
                }
            ]]}, {includes = "SFML/Graphics.hpp"}))
        end
```

**Comment by SirLynix** [comment]
Yes, I understood that it's a hidden issue my additional tests uncovered. But I don't understand why it's triggered on Linux with GCC but not with Clang, also libXext is linked.

**Comment by waruqi** [comment]
I don't know, it should work.

**Comment by waruqi** [comment]
> /usr/sbin/ld: cannot find -lXrender: No such file or directory


add Xrender dep?

File: packages/s/sfml/xmake.lua [hunk 1]
```diff
@@ -61,31 +61,31 @@ package("sfml")
...
         end
         if package:config("window") or package:config("graphics") then
             package:add("links", a .. "window" .. e)
-            if package:is_plat("windows", "mingw") then
+            if package:is_plat("windows", "mingw") and not package:config("shared") then
                 package:add("syslinks", "opengl32", "gdi32", "user32", "advapi32")
             end
             if package:is_plat("linux") then
-                package:add("deps", "libx11", "libxrandr", "freetype", "eudev")
+                package:add("deps", "libx11", "libxext", "libxrandr", "freetype", "eudev")
```
Comment (waruqi) [inline packages/s/sfml/xmake.lua:74]: add libxrender

**Pull request merged**